### DISCUSSION TITLE
Is it recommended to version control schema_versions.dart ?

### DISCUSSION THREAD

**Discussion by realth000** [simolus3/drift#3283] [Q&A] [answered]
Seems no one asked before and I didn't find information about this in doc.

In migration all `drift_schema_vX.json` files are recorded in git, so I think the `schema_versions.dart` file is reproducible.

Maybe there are bugfixes or better performance in the generated file carried by future drift versions, though unlikely.

And in my repo my db in schema version 4 binded with a 400 loc `schema_versions.dart` file, it's not quite much code but still has some.

I plan to exclude `schema_versions.dart` in git, is it recommended to do so?

**Comment by dickermoshe** [comment]
Yes

**Reply by dickermoshe** [reply]
More Info:

When you want to write migrations and test those migrations, those files will help you tremendously.
New tooling for this has just been merged and should be coming out

**Reply by simolus3** [reply, marked as answer]
There's nothing wrong with excluding or including that file. As you've said, drift re-writes the file every time you invoke the migration command, so there's not a big difference. I include that file (and most generated sources) in my projects so that I can track changes in case anything goes wrong, but it's mostly a personal preference and I don't see any potential issues when excluding the file.

**Reply by dickermoshe** [reply]
If you aren't going to check it into version control, then what's the point of generating it in the 1st place?

**Reply by simolus3** [reply]
> then what's the point of generating it in the 1st place?

Well, you could generate these files on demand (like we do for some of the generated files in the documentation).

**Reply by realth000** [reply]
> More Info:
> 
> When you want to write migrations and test those migrations, those files will help you tremendously.
> New tooling for this has just been merged and should be coming out
> 

Thank you. I saw the new tooling PR, it's great.

**Reply by realth000** [reply]
> There's nothing wrong with excluding or including that file. As you've said, drift re-writes the file every time you invoke the migration command, so there's not a big difference. I include that file (and most generated sources) in my projects so that I can track changes in case anything goes wrong, but it's mostly a personal preference and I don't see any potential issues when excluding the file.

Thanks, I will include it in git so that I could run one less command after `flutter clean` each time (laughing).

**Reply by dickermoshe** [reply]
@simolus3 Oh, I thought he was asking about the `*.json` files.
Yeah, you can delete those files. The CLI generates them pretty quickly, so you don't have to check them in.

(note to self: don't answer questions once you've taken out your contact lenses)
